core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/internal/UserStoreMgtDataHolder.java

```diff
@@ -47,4 +52,21 @@ public CryptoService getCryptoService() {
         return cryptoService;
     }
 
+    public void setHashProvider(HashProvider hashProvider) {
+
+        if (hashProviderMap == null) {
+            hashProviderMap = new HashMap<>();
+        }
+        hashProviderMap.put(hashProvider.getAlgorithm(), hashProvider);
+    }
+
+    public HashProvider getHashProvider(String algorithm) {
+
+        return hashProviderMap.get(algorithm);
```

Comment:
Better if we can validate `hashProviderMap` being NULL

Reply:
I think it is better to validate the key-value pair, whether it is available or not. Validating an empty map will also be included in validating the key-value pairs.

Reply:
```private void validateHashProviderMap(String algorithm) throws HashProviderException{

        if (hashProviderMap.get(algorithm) == null) {
            throw new HashProviderException("Specified hashing algorithm is not available.");
        }
    }```
like this

Reply:
```
private void validateHashProviderMap(String algorithm) throws HashProviderException{

        if (hashProviderMap.get(algorithm) == null) {
            throw new HashProviderException("Specified hashing algorithm is not available.");
        }
    }
```

Reply:
But then need to throw HashProviderException at preparePassword method itself.